File: spec/support/single_object_dvd/linux/interfaces/php/IpLocatorSample.php

<?php

#********************************************************************************
#If you should have any comments, suggestions or improvements to these samples,
#we welcome you to contact us at SampleCode@melissadata.com also please visit our
#developers bulletin board at forum.melissadata.com.
#********************************************************************************

#! /usr/bin/php


include './mdIpLocatorPHPWrapper.php';

#Put License and Directory of IpLocator Object Data Files here

//********************** LICENSE STRINGS ***********************
//*     To unlock the functionality of IpLocator Object,      *
//*     	     You must have a license string.           	*
//* Please call a sales representative at 1-800-MELISSA ext. 3 *
//*         (1-800-635-4772 x3) for a license string.          *
//**************************************************************/

$License = "License Here";
$DataPath = "/DQT/Current";

#Create New Object
$ipObj = new mdIpLocator();

#Set License String
$ipObj->SetLicenseString($License);

#Set Data Path and Initialize
$ipObj->SetPathToIpLocatorFiles($DataPath);

if($ipObj->InitializeDataFiles() != 0)
{
	print "Cannot Initialize: ".$ipObj->GetInitializeErrorString()."\n";
	exit;
}

# Print Initialized Information
print "Build:".$ipObj->GetBuildNumber()."\n";
print "Database Date: ".$ipObj->GetDatabaseDate()."\n";
print "Database Expiration Date: ".$ipObj->GetDatabaseExpirationDate()."\n";
print "License Expiration Date: ".$ipObj->GetLicenseStringExpirationDate()."\n";

#Get Input
print "Enter IP Address: ";
$ipaddress = trim(fgets(STDIN));

$ipObj->LocateIpAddress($ipaddress);

if(strpos($ipObj->GetResults(),"IS01") !== false)
{
	print "Latitude: " . $ipObj->GetLatitude() . "\n";
	print "Longitude: ". $ipObj->GetLongitude() . "\n";
	print "City: " . $ipObj->GetCity() . "\n";
	print "Region: " . $ipObj->GetRegion() . "\n";	
	print "Zip: " . $ipObj->GetZip() . "\n";
      print "Country: " . $ipObj->GetCountry() . "\n";
      print "Country Abbreviation: " . $ipObj->GetCountryAbbreviation() . "\n";
 	print "Domain Name: " . $ipObj->GetDomainName() . "\n";
	print "ISP: " . $ipObj->GetISP() . "\n";
}
elseif(strpos($ipObj->GetResults(),"IS02") !== false)
{
	# Display results
	print "IP Address not found. \n";
}
elseif(strpos($ipObj->GetResults(),"IE01") !== false)
{
	# Display results
	print "IP Address empty or not well formed. \n";
}

?>

File: spec/support/single_object_dvd/linux/interfaces/python/IpLocatorSample.py
# ********************************************************************************
# If you should have any comments, suggestions or improvements to these samples, 
# we welcome you to contact us at SampleCode@melissadata.com also please visit our 
# developers bulletin board at forum.melissadata.com.
# ********************************************************************************

import mdIpLocatorPythonWrapper

#/********************** LICENSE STRINGS **********************
#*     To unlock the full functionality of IpLocator Object,    *
#* please call a sales representative at 1-800-MELISSA ext. 3 *
#*         (1-800-635-4772 x3) for a license string.          *
#*  IpLocator Object will not function without a valid *
#*      			  license string 			          *
#**************************************************************/

# Put your License String and Path to IpLocator Object Data Files here
License = r"License Here"
DataPath = r"/DQT/Current"


# Create IpLocator Object
ipObj = mdIpLocatorPythonWrapper.mdIpLocator()

# Set License String
ipObj.SetLicenseString(License)
    
# Initialize Data Files
ipObj.SetPathToIpLocatorFiles(DataPath)

result = ipObj.InitializeDataFiles()

if result != 0:
    print "Initialize Error: ", ipObj.GetInitializeErrorString()
    exit()

print "======================================"
print "BuildNumber: ", ipObj.GetBuildNumber() 
print "Database Date: ", ipObj.GetDatabaseDate()
print "Database Expiration Date: ", ipObj.GetDatabaseExpirationDate()
print "License Expiration Date: ", ipObj.GetLicenseStringExpirationDate()
print "======================================\n"

while(1):
    ipaddress = raw_input("Enter IP Addres: ")
    
    ipObj.LocateIpAddress(ipaddress)

    if ipObj.GetResults().find('IS01') != -1:
        print "Latitude: " , ipObj.GetLatitude()
        print "Longitude: " , ipObj.GetLongitude()
        print "City: " , ipObj.GetCity()
        print "Region: " , ipObj.GetRegion()
	print "Zip: " , ipObj.GetZip()
        print "Country: " , ipObj.GetCountry()
        print "Country Abbreviation: " , ipObj.GetCountryAbbreviation()
	print "Domain Name: " , ipObj.GetDomainName()
        print "ISP: " ,ipObj.GetISP()
        print "\n"
    elif ipObj.GetResults().find('IS02') != -1:
        print "IP Address not found"
    elif ipObj.GetResults().find('IE01') != -1:
        print "IP Address empty or not well formed"
       
        print "\n"


